core/services/pipeline/scheduler.go

```diff
@@ -165,8 +165,7 @@ func (s *scheduler) Run() {
 
 		s.waiting--
 
-		// TODO: this is temporary until task_bridge can return a proper pending result
-		if result.Result.Error == ErrPending {
+		if result.runInfo.IsPending {
```

Comment:
You can probably remove this whole if and replace `result.IsPending()` on L185 with `result.runInfo.IsPending`.